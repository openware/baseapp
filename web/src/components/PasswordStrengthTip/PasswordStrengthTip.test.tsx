import { shallow } from 'enzyme';
import * as React from 'react';
import { PasswordStrengthTip, PasswordStrengthTipProps } from './';

const defaults: PasswordStrengthTipProps = {
    passwordErrorFirstSolved: false,
    passwordErrorSecondSolved: false,
    passwordErrorThirdSolved: false,
    passwordPopUp: false,
    translate: jest.fn(),
};

const setup = (props: Partial<PasswordStrengthTipProps> = {}) =>
    shallow(<PasswordStrengthTip {...{...defaults, ...props }} />);

describe('PasswordStrengthMeter component', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
    });
});
