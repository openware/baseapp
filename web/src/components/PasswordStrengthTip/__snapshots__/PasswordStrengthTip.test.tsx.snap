// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordStrengthMeter component should render 1`] = `
<div
  className="pg-password-strength-tip"
>
  <span
    className="pg-password-strength-tip-title"
  />
  <span
    className="pg-password-strength-tip-text"
  />
  <span
    className="pg-password-strength-tip-text"
  />
  <span
    className="pg-password-strength-tip-text"
  />
</div>
`;
